src/libgit2/repository.c

```diff
@@ -1147,15 +1147,24 @@ int git_repository_open_from_worktree(git_repository **repo_out, git_worktree *w
 
 int git_repository_wrap_odb(git_repository **repo_out, git_odb *odb)
 {
-	git_repository *repo;
+	int error;
+	git_repository *repo = NULL;
+	git_config *config = NULL;
 
 	repo = repository_alloc();
 	GIT_ERROR_CHECK_ALLOC(repo);
 
 	git_repository_set_odb(repo, odb);
+
+	if ((error = obtain_config_and_set_oid_type(&config, repo)) < 0)
```

Comment:
Interesting. I was going to argue that this was not necessary because a fake repository would never have a config of its own. But then I poked around to validate this thinking. This made me realize that a) git limits the repository format version config and / or per-repo extensions to the repo config and doesn't look at system / global. and b) we do not do this.

That's something that we should fix independently of this problem. But it means that we can simplify this to not bother looking at the config. So I think that the simpler solution is just to `repo->oid_type = GIT_OID_DEFAULT;`...

But I think that in the long term, though, this function needs to take an oid type. I think that I might do:

```c
#ifdef GIT_EXPERIMENTAL_SHA256
int git_repository_wrap_odb(git_repository **repo_out, git_odb *odb, git_oid_t oid_type)
#else
int git_repository_wrap_odb(git_repository **repo_out, git_odb *odb)
#endif
{
#ifndef GIT_EXPERIMENTAL_SHA256
	git_oid_t oid_type = GIT_OID_DEFAULT;
#endif

	repo = repository_alloc();
	GIT_ERROR_CHECK_ALLOC(repo);

	repo->oid_type = oid_type;

	git_repository_set_odb(repo, odb);

	*repo_out = repo;
	return 0;
}
```

Reply:
Alright, I updated the branch to implement the simple solution :)